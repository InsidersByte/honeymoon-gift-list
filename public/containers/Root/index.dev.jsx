/* @flow */

import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import DevTools from '../DevTools';
import routes from '../../routes';

type PropsType = {
    store: Object,
    history: Object,
};

const styles = {
    root: {
        height: '100%',
        width: '100%',
    },
};

const Root = ({ store, history }: PropsType) => (
    <Provider store={store}>
        <div style={styles.root}>
            <Router history={history}>
                {routes(store)}
            </Router>

            <DevTools />
        </div>
    </Provider>
);

export default Root;
