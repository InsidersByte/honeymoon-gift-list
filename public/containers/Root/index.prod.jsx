/* @flow */

import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import routes from '../../routes';

type PropsType = {
    store: Object,
    history: Object,
};

const Root = ({ store, history }: PropsType) => (
    <Provider store={store}>
        <Router history={history}>
            {routes(store)}
        </Router>
    </Provider>
);

export default Root;
