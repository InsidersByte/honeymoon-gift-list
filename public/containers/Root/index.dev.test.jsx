import React from 'react';
import { shallow } from 'enzyme';
import Root from './index.dev';

describe('Root', () => {
    it('should render correctly', () => {
        const wrapper = shallow(
            <Root>
                <h1>Hello World</h1>
            </Root>,
        );

        expect(wrapper).toMatchSnapshot();
    });
});
