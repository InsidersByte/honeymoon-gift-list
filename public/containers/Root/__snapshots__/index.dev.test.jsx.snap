// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root should render correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "liftedStore": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <div
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <Router
      history={Object {}}
      render={[Function]}
    >
      <Route
        component={[Function]}
        path="/"
      >
        <Route
          component={[Function]}
        >
          <IndexRoute
            component={[Function]}
          />
          <Route
            component={[Function]}
            path="basket"
          />
          <Route
            component={[Function]}
            path="giver"
          />
          <Route
            component={[Function]}
            path="confirmation/:giftSetId"
          />
        </Route>
        <Route
          component={[Function]}
          path="admin"
        >
          <IndexRedirect
            to="giftSet"
          />
          <Route
            component={[Function]}
            onEnter={[Function]}
          >
            <Route
              component={[Function]}
              onEnter={[Function]}
              path="setup"
            />
            <Route
              onEnter={[Function]}
            >
              <Route
                component={[Function]}
                path="login"
              />
              <Route
                component={[Function]}
                path="reset/:token"
              />
              <Route
                component={[Function]}
                path="signUp/:token"
              />
            </Route>
          </Route>
          <Route
            onEnter={[Function]}
          >
            <Route
              component={[Function]}
              onEnter={[Function]}
            >
              <Route
                component={[Function]}
                path="profile"
              />
              <Route
                component={[Function]}
                path="weddingProfile"
              />
              <Route
                component={[Function]}
                path="gift"
              />
              <Route
                component={[Function]}
                path="section"
              />
              <Route
                component={[Function]}
                path="section/create"
              />
              <Route
                component={[Function]}
                path="section/:id"
              />
              <Route
                component={[Function]}
                path="user"
              />
              <Route
                component={[Function]}
                path="giftSet"
              />
              <Route
                component={[Function]}
                path="giftSet/:giftSetId"
              />
              <Route
                component={[Function]}
                path="weddingPartyMember"
              />
              <Route
                component={[Function]}
                path="weddingPartyMember/create"
              />
              <Route
                component={[Function]}
                path="weddingPartyMember/:id"
              />
              <Route
                component={[Function]}
                path="*"
              />
            </Route>
          </Route>
        </Route>
        <Route
          component={[Function]}
          path="*"
        />
      </Route>
    </Router>
    <DevTools />
  </div>
</Provider>
`;
